Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnSlide.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class OnSlide : OnMovementSpecial
+{
+
+    [SerializeField]
+    private float wallSlideUpForce;
+    [SerializeField]
+    private float wallSlideDownForce;
+    [SerializeField]
+    private float wallSlideUpDecay;
+    [SerializeField]
+    private float wallSlideDownDecay;
```

Comment:
nit: decay is a misleading name for this, since it actually makes the player's speed increase